[flang] Modifications to ieee_support_standard

vdonaldson opened the pull request:
Some Arm processors support exception halting control and some do not. An Arm executable will run on either type of processor, so it is effectively unknown at compile time whether or not this support will be available. ieee_support_halting is therefore implemented with a runtime check.

The result of a call to ieee_support_standard depends in part on support for halting control. Update the ieee_support_standard implementation to check for support for halting control at runtime.

vdonaldson committed:
[flang] Modifications to ieee_support_standard

Some Arm processors support exception halting control and some do not.
An Arm executable will run on either type of processor, so it is effectively
unknown at compile time whether or not this support will be available.
ieee_support_halting is therefore implemented with a runtime check.

The result of a call to ieee_support_standard depends in part on support
for halting control. Update the ieee_support_standard implementation to
check for support for halting control at runtime.

tblah commented:
Hi, thank you for the patch.

After this patch, two Fujitsu tests now fail to build: 0136_0046.f90 and 0136_0047.f90. The error message complains that `ieee_support_standrard(x) must be a scalar value, but is a rank-1 array`.

vdonaldson committed:
[flang] review update

Procedure ieee_support_standard is an inquiry function, not an elemental
function. Account for that, and make similar adjustments for other inquiry
and transformational functions.

vdonaldson commented:
> After this patch, two Fujitsu tests now fail to build: 0136_0046.f90 and 0136_0047.f90. The error message complains that `ieee_support_standrard(x) must be a scalar value, but is a rank-1 array`.

Thanks for catching this. I've updated the code to treat `ieee_support_standard` and similar procedures as inquiry or transformational functions, as appropriate, rather than as elemental functions.

tblah reviewed (approved):
Thanks for the update. This resolves the issue I mentioned.
